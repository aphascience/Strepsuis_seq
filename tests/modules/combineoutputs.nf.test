nextflow_process {

    name "Test Process COMBINE_OUTPUTS"
    script "modules/combineoutputs.nf"
    process "COMBINE_OUTPUTS"

    test("Should run without failures") {

        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                input[0] = file("${projectDir}/tests/data/recN_table.tsv")
                input[1] = file("${projectDir}/tests/data/mlst_table.tsv")
                input[2] = file("${projectDir}/tests/data/serotype_table.tsv")
                input[3] = file("${projectDir}/tests/data/virulence_table.tsv")
                input[4] = file("${projectDir}/tests/data/verification.csv")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
