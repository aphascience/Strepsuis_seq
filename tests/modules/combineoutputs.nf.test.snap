{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "tests_FinalOut_24Sep25.csv:md5,feed601f74d28ed2abfefa99c15635dc"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-09-24T15:19:39.078051376"
    }
}